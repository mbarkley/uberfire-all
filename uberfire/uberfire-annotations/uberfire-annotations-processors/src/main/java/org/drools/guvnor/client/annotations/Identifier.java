package org.drools.guvnor.client.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import javax.inject.Qualifier;

/**
 * An identifier for a Place (WorkbenchPart or Popup)
 */
@Qualifier
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE})
public @interface Identifier {

    String value();

}