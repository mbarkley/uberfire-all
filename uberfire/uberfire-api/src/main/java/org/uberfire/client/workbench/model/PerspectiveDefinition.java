/*
 * Copyright 2012 JBoss Inc
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package org.uberfire.client.workbench.model;

/**
 * Meta-data defining a Perspective. A Perspective is a set of Panels and Parts
 * arranged within the Workbench. One Workbench contains multiple Panels, each
 * Panel contains multiple Parts. Each Part contains one PlaceRequest.
 */
public interface PerspectiveDefinition {

    /**
     * Get the name of the Perspective.
     * 
     * @return The name of the Perspective.
     */
    public String getName();

    /**
     * Set the name of the Perspective.
     * 
     * @param name
     *            The name of the Perspective.
     */
    public void setName(final String name);

    /**
     * Get the root Panel for this Perspective. The root Panel contains all
     * child Panels. A Perspective is based on a single root Panel.
     * 
     * @return The root Panel.
     */
    public PanelDefinition getRoot();

}
