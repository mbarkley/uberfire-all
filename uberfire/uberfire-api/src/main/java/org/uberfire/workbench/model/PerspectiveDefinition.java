/*
 * Copyright 2012 JBoss Inc
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package org.uberfire.workbench.model;

/**
 * Meta-data defining a Perspective. A Perspective is a set of Panels and Parts
 * arranged within the Workbench. One Workbench contains multiple Panels, each
 * Panel contains multiple Parts. Each Part contains one PlaceRequest.
 */
public interface PerspectiveDefinition {

    /**
     * Get whether the Perspective is transient, i.e. will not be persisted.
     * @return True if the Perspective is transient and is not to be persisted.
     */
    public boolean isTransient();

    /**
     * Set whether the Perspective is transient, i.e. will not be persisted.
     * @param isTransient True if the Perspective is not to be persisted.
     */
    public void setTransient( final boolean isTransient );

    /**
     * Get the name of the Perspective.
     * @return The name of the Perspective.
     */
    public String getName();

    /**
     * Set the name of the Perspective.
     * @param name The name of the Perspective.
     */
    public void setName( final String name );

    /**
     * Get the root Panel for this Perspective. The root Panel contains all
     * child Panels. A Perspective is based on a single root Panel.
     * @return The root Panel.
     */
    public PanelDefinition getRoot();

    void setContextDefinition( final ContextDefinition contextDefinition );

    ContextDefinition getContextDefinition();

    ContextDisplayMode getContextDisplayMode();

    void setContextDisplayMode( final ContextDisplayMode contextDisplayMode );
}
