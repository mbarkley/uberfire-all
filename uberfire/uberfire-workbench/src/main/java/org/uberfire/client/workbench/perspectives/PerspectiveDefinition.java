/*
 * Copyright 2012 JBoss Inc
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package org.uberfire.client.workbench.perspectives;

import java.util.HashSet;
import java.util.Set;

/**
 * Meta-data defining a Perspective. A Perspective is a set of WorkbenchPanels
 * and WorkbenchParts arranged within the Workbench. Each WorkbenchPart
 * containing a PlaceRequest. Perspectives can be persisted.
 */
public class PerspectiveDefinition {

    private String name;

    private Set<PerspectivePartDefinition> parts = new HashSet<PerspectivePartDefinition>();

    public String getName() {
        return name;
    }

    public void setName(final String name) {
        this.name = name;
    }

    public void addPart(final PerspectivePartDefinition part) {
        parts.add( part );
    }

    public Set<PerspectivePartDefinition> getParts() {
        return this.parts;
    }

}
