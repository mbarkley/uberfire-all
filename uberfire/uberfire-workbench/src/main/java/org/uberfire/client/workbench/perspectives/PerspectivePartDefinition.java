/*
 * Copyright 2012 JBoss Inc
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package org.uberfire.client.workbench.perspectives;

import java.util.HashSet;
import java.util.Set;

import org.uberfire.client.workbench.Position;
import org.uberfire.shared.mvp.PlaceRequest;

/**
 * 
 */
public class PerspectivePartDefinition {

    private Set<PerspectivePartDefinition> parts = new HashSet<PerspectivePartDefinition>();

    private Position                       position;
    private PlaceRequest                   place;

    public PerspectivePartDefinition() {
    }

    public PerspectivePartDefinition(final Position position,
                                     final PlaceRequest place) {
        this.position = position;
        this.place = place;
    }

    public void addPart(final PerspectivePartDefinition part) {
        parts.add( part );
    }

    public Set<PerspectivePartDefinition> getParts() {
        return this.parts;
    }

    public Position getPosition() {
        return position;
    }

    public PlaceRequest getPlace() {
        return place;
    }

}
