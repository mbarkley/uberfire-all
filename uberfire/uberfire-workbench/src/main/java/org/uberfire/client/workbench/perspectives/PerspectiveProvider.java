/*
 * Copyright 2012 JBoss Inc
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package org.uberfire.client.workbench.perspectives;

import org.uberfire.client.workbench.widgets.panels.PanelManager;

/**
 * A Perspective Provider sets the default layout of the Workbench.
 */
public interface PerspectiveProvider {

    /**
     * The name of the perspective
     * 
     * @return
     */
    String getName();

    /**
     * Called by the Workbench for the Perspective Provider to build the
     * applicable panels etc. Implementations should add WorkbenchPanels to the
     * root panel obtained by calling panelManager.getRoot(). All existing
     * panels will be closed.
     * 
     * @param panelManager
     */
    void buildWorkbench(final PanelManager panelManager);

}
