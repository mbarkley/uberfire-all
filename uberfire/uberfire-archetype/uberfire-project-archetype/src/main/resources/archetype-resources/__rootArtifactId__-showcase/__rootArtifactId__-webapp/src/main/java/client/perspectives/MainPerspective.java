#set( $symbol_pound = '#' )
#set( $symbol_dollar = '$' )
#set( $symbol_escape = '\' )
package ${package}.client.perspectives;

import javax.enterprise.context.ApplicationScoped;

import org.uberfire.client.annotations.Perspective;
import org.uberfire.client.annotations.WorkbenchPerspective;
import org.uberfire.mvp.impl.DefaultPlaceRequest;
import org.uberfire.workbench.model.PanelType;
import org.uberfire.workbench.model.PerspectiveDefinition;
import org.uberfire.workbench.model.impl.PartDefinitionImpl;
import org.uberfire.workbench.model.impl.PerspectiveDefinitionImpl;

@ApplicationScoped
@WorkbenchPerspective(identifier = "MainPerspective", isDefault = true)
public class MainPerspective {

    @Perspective
    public PerspectiveDefinition buildPerspective() {
        final PerspectiveDefinition p = new PerspectiveDefinitionImpl( PanelType.ROOT_LIST );
        p.setTransient( true );
        p.setName( "MainPerspective" );

        p.getRoot().addPart( new PartDefinitionImpl( new DefaultPlaceRequest( "HelloWorldScreen" ) ) );
        p.getRoot().addPart( new PartDefinitionImpl( new DefaultPlaceRequest( "ComponentPresenter" ) ) );

        return p;
    }
}
