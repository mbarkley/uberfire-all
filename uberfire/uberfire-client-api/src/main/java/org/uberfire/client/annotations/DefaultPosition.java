/*
 * Copyright 2012 JBoss Inc
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 */
package org.uberfire.client.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.uberfire.workbench.model.Position;

import static org.uberfire.workbench.model.Position.ROOT;

/**
 * Default position for WorkbenchParts added to the Workbench. Methods marked
 * with this annotation should take zero arguments and return
 * {@code org.drools.guvnor.client.workbench.Position} indicating the
 * WorkbenchPart's default position.
 */
@Inherited
@Retention(RetentionPolicy.RUNTIME)
@Target({ ElementType.METHOD })
public @interface DefaultPosition {

    Position getPosition() default ROOT;

}
