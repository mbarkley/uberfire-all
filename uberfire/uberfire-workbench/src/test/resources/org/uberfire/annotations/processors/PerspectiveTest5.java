package org.uberfire.annotations.processors;

import org.uberfire.client.annotations.Perspective;
import org.uberfire.client.workbench.perspectives.PerspectiveDefinition;
import org.uberfire.security.annotations.All;
import org.uberfire.security.annotations.Roles;

public class PerspectiveTest5 {

    @Perspective(identifier = "PerspectiveTest5", isDefault = true)
    @All @Roles({"ADMIN", "SUDO"})
    public PerspectiveDefinition getPerspective() {
        return null;
    }

}
