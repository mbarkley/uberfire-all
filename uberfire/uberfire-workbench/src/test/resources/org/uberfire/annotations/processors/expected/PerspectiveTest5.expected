/*
 * Copyright 2012 JBoss Inc
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.uberfire.annotations.processors;

import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import javax.annotation.Generated;
import javax.enterprise.context.ApplicationScoped;
import javax.inject.Inject;

import org.uberfire.client.mvp.AbstractPerspectiveActivity;
import org.uberfire.client.mvp.PlaceManager;

import org.uberfire.client.workbench.model.PerspectiveDefinition;
import org.uberfire.client.workbench.annotations.Identifier;
import org.uberfire.client.workbench.annotations.DefaultPerspective;

@ApplicationScoped
@DefaultPerspective
@Generated("org.uberfire.annotations.processors.PerspectiveProcessor")
@Identifier("PerspectiveTest5")
/*
 * WARNING! This class is generated. Do not modify.
 */
public class PerspectiveTest5Activity_getPerspective extends AbstractPerspectiveActivity {

    private static final Collection<String> ROLES = Arrays.asList("ADMIN", "SUDO");

    private static final Collection<String> TRAITS = Arrays.asList("org.uberfire.security.annotations.All");

    @Inject
    private PerspectiveTest5 realClass;

    @Inject
    //Constructor injection for testing
    public PerspectiveTest5Activity_getPerspective(final PlaceManager placeManager) {
        super( placeManager );
    }

    @Override
    public String getIdentifier() {
        return "PerspectiveTest5";
    }

    @Override
    public PerspectiveDefinition getPerspective() {
        return realClass.getPerspective(); 
    }
    
    @Override
    public Collection<String> getRoles() {
        return ROLES;
    }

    @Override
    public Collection<String> getTraits() {
        return TRAITS;
    }

    @Override
    public String getSignatureId() {
        return "org.uberfire.annotations.processors.PerspectiveTest5Activity_getPerspective";
    }
}
